tests/obj_factory_test.cpp

```diff
@@ -0,0 +1,99 @@
+/* Copyright © 2001-2015, Canal TP and/or its affiliates. All rights reserved.
+  
+This file is part of Navitia,
+    the software to build cool stuff with public transport.
+ 
+Hope you'll enjoy and contribute to this project,
+    powered by Canal TP (www.canaltp.fr).
+Help us simplify mobility and open public transport:
+    a non ending quest to the responsive locomotion way of traveling!
+  
+LICENCE: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+   
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+   
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+  
+Stay tuned using
+twitter @navitia 
+IRC #navitia on freenode
+https://groups.google.com/d/forum/navitia
+www.navitia.io
+*/
+
+#include "utils/obj_factory.h"
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE obj_factory_test
+#include <boost/test/unit_test.hpp>
+#include "utils/functions.h"
+
+struct HeaderInt {
+    uint32_t idx;
+    std::string uri;
+    int val = 0;
+};
+
+BOOST_AUTO_TEST_CASE(simple_obj_factory) {
+    auto i_int = HeaderInt();
+    i_int.val = 2;
+    auto iInt_bis = i_int;
+    navitia::ObjFactory<HeaderInt> obj_factory;
+
+    obj_factory.emplace("Hubert"); //name dropping
+    BOOST_CHECK_EQUAL(obj_factory["Hubert"]->val, 0);
+    obj_factory.get_or_create("Hubert", i_int); //should not change value
+    BOOST_CHECK_EQUAL(obj_factory[navitia::Idx<HeaderInt>(0)]->val, 0);
+    obj_factory.get_or_create("Hubert")->val = 1; //should change value
+    BOOST_CHECK_EQUAL(obj_factory.get_mut(navitia::Idx<HeaderInt>(0))->val, 1);
```

Comment:
To test that all getters are working. Did not try to challenge constness on all of them though... Seems enough to me.